### PR TITLE
Fix upload data button permission

### PR DESCRIPTION
**Pull request by danangmassandy**
This is for #1772.

Upload Data link is disabled for Data Consumer/Data Scientiest:
image

There is also validation when accessing the /upload or /map?tab=4 to raise permission denied if this URL is accessed by Data Consumer/Data Scientiest.
I also hide the actions column in Species Report:
image

### CODE REVIEW

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 8 min 10 sec

**Review by dimasciput — CHANGES_REQUESTED** [review]

File: django_project/frontend/src/utils/Helpers.ts [hunk 1]
```diff
@@ -60,3 +62,36 @@ export const isMapDisplayed = () => {
...
+  return userInfo.user_roles.some(userRole => dataConsumers.has(userRole))
+}
+
+
+/**
+ * Check if user has a Data Scientiest role
+ * @param userInfo user info from API
+ * @returns True if user has group/role of Data Consumer
+ */
+export function isDataScientiest(userInfo: UserInfo) {
```
[the remark this answers is not in the document]
Comment (danangmassandy) [reply django_project/frontend/src/utils/Helpers.ts]: updated

**Review by dimasciput — APPROVED** [review]

**Comment by dimasciput** [comment]
@danangmassandy please fix the conflicts

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 8 min 37 sec

**Comment by danangmassandy** [comment]
I'll merge this PR, the test that was failed is playwright test.

**Comment by lgkgh** [comment]
#### Jenkins Build Details:
**Build status:** SUCCESS
**Build duration:** 2 min 54 sec

**Pull request merged**